github.md

```diff
@@ -0,0 +1,145 @@
+---
+title: GitHub Contributions
+parent: Contributing
+nav_order: 4
+layout: default
+---
+
+# Overview
+
+GitHub handles contributions as [pull
+requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) to
+relevant repositories part of the [Eclipse Oniro for OpenHarmony organization](https://github.com/eclipse-oniro4openharmony).
+The flow for handling that is classic: fork-based pull requests. This means that
+once you have an account, you can fork any repository, create a branch
+with proposed changes and raise a pull request against the forked
+repository. More generic information you can find on the GitHub\'s
+documentation as part of [\"Working with forks\"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks).
+
+# Git setup
+
+Clone your fork locally, enter its directory and set:
+
+```bash
+$ git config --local user.email <your_eclipse_account_email>
+$ git config --local user.name <your_eclipse_full_name>
+```
+
+If you want to push or pull repositories using SSH, you have to [add a
+SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) to your profile.
+
+# Commit Guidelines
+
+> 💡 **NOTE：** If you are new to `git`, start by reading the official [Getting Started Document](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup).
+
+At its core, contributing to the project means _wrapping_ your work as
+`git` commits. How we handle this has an impact on rebasing,
+cherry-picking, back-porting, and ultimately exposing consistent
+documentation through its logs.
+
+To achieve this, we maintain the following commit guidelines:
+
+- Each commit should be able to stand by itself providing a building
+  block as part of the pull request (PR).
+  - A good balance of granularity with scoped commits helps to
+    handle backports (e.g. cherry-picks) and also improves the
+    ability to review smaller chunks of code taking commit by
+    commit.
+- Changes that were added on top of changes introduced in the PR,
+  should be squashed into the initial commit.
+  - For example, a PR that introduced a new build system recipe and,
+    as a separate commit, fixed a build error in the initial recipe.
```

Comment:
this example refers to yocto build system terminologies

Reply:
Good point. Changed it to a more generic feature. Fixed.